commandapi-core/src/main/java/dev/jorel/commandapi/network/packets/ClientToServerUpdateRequirementsPacket.java

```diff
@@ -0,0 +1,75 @@
+package dev.jorel.commandapi.network.packets;
+
+import com.google.common.io.ByteArrayDataInput;
+import com.google.common.io.ByteArrayDataOutput;
+import com.mojang.brigadier.builder.ArgumentBuilder;
+import dev.jorel.commandapi.network.CommandAPIMessenger;
+import dev.jorel.commandapi.network.CommandAPIPacket;
+import dev.jorel.commandapi.network.CommandAPIPacketHandler;
+
+import java.util.function.Predicate;
+
+/**
+ * A packet that signals a client wants to be sent the <a href="https://wiki.vg/Protocol#Commands">Commands packet</a>.
+ * <p>
+ * This packet does not contain any data. The receiver should simply update the requirements for the client the packet
+ * came from.
+ * <p>
+ * When the Commands packet is resent, Brigadier reevaluates if that player is able to run each command. If the
+ * conditions tested by {@link ArgumentBuilder#requires(Predicate)} have changed, the client is informed of those
+ * changes, updating what they know about their requirements.
+ */
+public class ClientToServerUpdateRequirementsPacket implements CommandAPIPacket {
```

Comment:
This name is mildly misleading. "Client" typically refers to the Notchian client (i.e. a player), and "Server" typically refers to a Minecraft server (e.g. Vanilla, Bukkit, Spigot, Paper).

If my understanding is correct, in this class naming scheme the "client" refers to the server, and "server" refers to the proxy?

Also, in traditional MCP (I believe), packets were named to indicate the direction of travel. Something such as "ClientToServer" implies that this packet will be sent from the client (Notchian client) to the server (Bukkit/Spigot/Paper server) which is incorrect (players don't request requirement updates, we explicitly give players requirement updates when we want).

Reply:
If there are two directions of travel, those would be `ClientToServer` and `ServerToClient`. In Velocity, those would be `Client -> Velocity -> Server` and `Server -> Velocity -> Client` respectively. Since this packet travels `Velocity -> Server`, that would be the first case, so I called it `ClientToServer`. I suppose you could have `ProxyToServer` and `ServerToProxy`, but I think C2S and S2C make sense anyway.

I was also thinking about how this system would be used for other platforms. Eventually, the CommandAPI could be ported to a client-side modding platform. When `CommandAPI.updateRequirements()` is called there, it could send a `ClientToServerUpdateRequirementsPacket` to get the base server to resend the Commands packet. In this case, C2S definitely makes sense, especially when there isn't a proxy in the middle.

Reply:
I just had a different idea:
Instead of calling it `ClientToServer` or `ProxyToServer` or whatever, why not name it what it actually does without specifying a direction since (to my understanding) the packets can be sent both ways.
So instead of calling it `ClientToServerUpdateRequirementsPacket`, why don't we just name it `UpdateRequirementsPacket`?

Reply:
Yeah, there is nothing stopping the update requirements packet from being sent towards the client. However, I don't think it would ever make sense to do that. This packet indicates that Minecraft's Commands packet should be resent. The Commands packet definitely only travels from server to client, so the update requirements packet really only makes sense traveling client to server. On the other hand, certain packets may make sense in both directions, so the lack of direction in the class name could mean that the packet can go either way.

Maybe the direction doesn't matter that much though, and it shouldn't be in the packet's name. It could just be explained in the javadocs.

Reply:
There are now packets that can be sent in either direction. When I made the `SetVersionPacket` and `ProtocolVersionTooOldPacket`, I didn't add anything to the name to indicate their intended direction of travel. It does mention in the javadocs that these packets could be sent either way.

`ClientToServerUpdateRequirementsPacket` is different from these since it should only be sent Client -> Server, and I guess that is currently reflected in the name.

So, there's just a little question about what the format for packet names should be. Right now, if a packet has a specific direction, put that at the start of the name. If the packet can go either way, don't put a direction.

The format could just not have direction in the name no matter what. The intended direction could always just be specified by the javadocs.